Feature/add put customers (#1188)
mercadopago/devsite-docs: guides/online-payments/web-tokenize-checkout/customers-and-cards.en.md

	...
}]
```

> ## Modificar un Customer

Reviewer: ```suggestion
## Modify a customer
```

> Para modificar un cliente es necesario enviar el `customer_id` y los campos que se quieran actualizar en una solicitud `HTTP PUT`. 

Reviewer: ```suggestion
To modify a client you need to send the `customer_id` and the fields you want to update in a `HTTP PUT` request. 
```

> Los campos que se pueden modificar de un cliente son:

Reviewer: ```suggestion
The fields that can be modified from a client are:
```

> | Atributo | Descripción |

Reviewer: ```suggestion
| Attribute | Description |
```

| -------- | ----------- |

> | `address` | Dirección. |

Reviewer: ```suggestion
| `address` | Address. |
```

| `default_address` | Dirección por defecto. |

> | `default_card` | Tarjeta por defecto. |

Reviewer: ```suggestion
| `default_card` | Default card. |
```

> | `description` | Descripción. |

Reviewer: ```suggestion
| `description` | Description. |
```

> | `email` | E-mail de la cuenta. |

Reviewer: ```suggestion
| `email` | Account e-mail. |
```

> | `first_name` | Nombre. |

Reviewer: ```suggestion
| `first_name` | First name. |
```

> | `last_name` | Apellido. |

Reviewer: ```suggestion
| `last_name` | Last name. |
```

> | `phone` | Teléfono registrado. |

Reviewer: ```suggestion
| `phone` | Registered phone. |
```

> | `identification` | Tipo de documento y número. |

Reviewer: ```suggestion
| `identification` | Document type and number. |
```

[[[
```php

<?php

  MercadoPago\SDK::setAccessToken("ENV_ACCESS_TOKEN");

  $customer = new MercadoPago\Customer();
  $customer->email = "user@user.com";
  $customer->first_name = "john";
  $customer->last_name = "wagner";
  $customer->phone = array("area_code" => "[FAKER][PHONE_NUMBER][AREA_CODE]", "number" => "001234567");
  $customer->identification = array("type" => "[FAKER][IDENTIFICATION][TYPE]", "number" => "12341234");
  $customer->default_address = "Home";
  $customer->address = array("zip_code" => "[FAKER][ADDRESS][ZIP_CODE]", "street_name" => "[FAKER][ADDRESS][STREET_NAME]", "street_number" => "2");

>   $customer->description = "Información del cliente";

Reviewer: ```suggestion
  $customer->description = "Customer information";
```

  $customer->default_card = "None";
  $customer->update();

?>

```
```node

var mercadopago = require('mercadopago');
mercadopago.configure({
    access_token: 'ENV_ACCESS_TOKEN'
});

var customer_data = { 
  "email": "test@test.com",
  "first_name": "john" ,
  "last_name": "wagner",
  "phone": {
    "area_code": "[FAKER][PHONE_NUMBER][AREA_CODE]",
    "number": "001234567"
  },
  "identification": {
    "type": "[FAKER][IDENTIFICATION][TYPE]",
    "number": "12341234"
  }, 
  "default_address": "Home",
  "address": {
    "zip_code": "[FAKER][ADDRESS][ZIP_CODE]",
    "street_name": "[FAKER][ADDRESS][STREET_NAME]",
    "street_number": "2"
  },

>   "description": "Información del cliente",

Reviewer: ```suggestion
  "description": "Customer information",
```

>   "description": "Información del cliente"

Reviewer: ```suggestion
  "description": "Customer information"
```

  "default_card": "None
 }

mercadopago.customers.update(customer_data).then(function (customer) {
 // code ...
});

```

```java

import com.mercadopago.*;
MercadoPago.SDK.configure("ENV_ACCESS_TOKEN");

Phone phone = new Phone();
phone.setAreaCode("[FAKER][PHONE_NUMBER][AREA_CODE]");
phone.setNumber("001234567");

DefaultAddress defaultAddress = new DefaultAddress();
defaultAddress.setZipCode("[FAKER][ADDRESS][ZIP_CODE]");
defaultAddress.setStreetName("[FAKER][ADDRESS][STREET_NAME]");
defaultAddress.setStreetNumber(2);

Identification identification = new Identification();
identification.setType("[FAKER][IDENTIFICATION][TYPE]");
identification.setNumber(12341234);

Customer customer = new Customer();
customer.setEmail("user@user.com");
customer.setFirstName("john");
customer.setLastName("wagner");
customer.setDefaultAddress("Home");
customer.setPhone(phone);
customer.setIdentification(identification);

> customer.setDescription("Información del cliente");

Reviewer: ```suggestion
customer.setDescription("Customer information");
```

> customer.setDescription("Información del cliente");

Reviewer: ```suggestion
customer.setDescription("Customer information");
```

> customer.setDescription("Información del cliente");

Reviewer: ```suggestion
customer.setDescription("Customer information");
```

customer.setDefaultCard("None")
cusotmer.setAddress(defaultAddress)
customer.update();

```
```ruby

require 'mercadopago'

sdk = Mercadopago::SDK.new('ENV_ACCESS_TOKEN')

customer_request = {
  email: 'user@user.com',
  first_name: 'john',
  last_name: 'wagner',
  default_address: 'home',
  phone: {
    area_code: '[FAKER][PHONE_NUMBER][AREA_CODE]',
    number: '001234567'
  },
  identification: {
    type: '[FAKER][IDENTIFICATION][TYPE]',
    number: '12341234'
  },
  address: {
    zip_code: '[FAKER][ADDRESS][ZIP_CODE]',
    street_name: '[FAKER][ADDRESS][STREET_NAME]',
    street_number: '2'
  },

>   description: 'Información del cliente',

Reviewer: ```suggestion
  description: 'Customer information',
```

  default_card: 'None'
}
customer_response = sdk.customer.update(customer_id ,customer_request)
customer = customer_response[:response]

```
```csharp

MercadoPagoConfig.AccessToken = "ENV_ACCESS_TOKEN";
var phoneRequest = new PhoneRequest
{
  AreaCode = "[FAKER][PHONE_NUMBER][AREA_CODE]",
  Number = "001234567"
};

var identificationRequest = new IdentificationRequest
{
  Type = "[FAKER][IDENTIFICATION][TYPE]",
  Number = "12341234"
};

var addressRequest = new AddressRequest
{
  ZipCode = "[FAKER][ADDRESS][ZIP_CODE]",
  StreetName = "[FAKER][ADDRESS][STREET_NAME]",
  StreetNumber = "2"
};

var customerRequest = new CustomerRequest
{
    Email = "test@test.com",
    FirstName = "john",
    LastName = "wagner",
    DefaultAddress = "home",

>     Description = "Información del cliente",

Reviewer: ```suggestion
    Description = "Customer information",
```

>     Description = "Información del cliente",

Reviewer: ```suggestion
    Description = "Customer information",
```

>     Description = "Información del cliente",

Reviewer: ```suggestion
    Description = "Customer information",
```

customer_data = {
  "email": 'user@user.com',
  "first_name": 'john',
  "last_name": 'wagner',
  "default_address": 'home',
  "phone": {
    "area_code": '[FAKER][PHONE_NUMBER][AREA_CODE]',
    "number": '001234567'
  },
  "identification": {
    "type": '[FAKER][IDENTIFICATION][TYPE]',
    "number": '12341234'
  },
  "address": {
    "zip_code": '[FAKER][ADDRESS][ZIP_CODE]',
    "street_name": '[FAKER][ADDRESS][STREET_NAME]',
    "street_number": '2'
  },
  "description": 'Información del cliente',
  "default_card": 'None'
}
customer_response = sdk.customer().update(customer_id, customer_data)
customer = customer_response["response"]

```
```curl

curl -X PUT \
    'https://api.mercadopago.com/v1/customers/{id}' \
    -H 'Authorization: Bearer ACCESS_TOKEN_ENV' \ 
    -d '{
  "email": "user@user.com",
  "first_name": "john",
  "last_name": "wagner",
  "address": {
    "zip_code": "[FAKER][ADDRESS][ZIP_CODE]",
    "street_name": "[FAKER][ADDRESS][STREET_NAME]",
    "street_number": "2"
  },
  "phone": {
    "area_code": "[FAKER][PHONE_NUMBER][AREA_CODE]",
    "number": "001234567"
  },
  "identification": {
    "type": "[FAKER][IDENTIFICATION][TYPE]",
    "number": "12341234"
  },

>   "description": "Información del cliente" 

Reviewer: ```suggestion
  "description": "Customer information" 
```

}'

```
]]]

Ejemplo de respuesta con el envío del `customer_id`:
```json
{
  "id": "xxxxxxxxxxxxxxxxxxxxx",
  "email": "user@user.com",
  "first_name": "john",
  "last_name": "wagner",
  "phone": {
    "area_code": "[FAKER][PHONE_NUMBER][AREA_CODE]",
    "number": 001234567
  },
  "identification": {
    "type": "[FAKER][IDENTIFICATION][TYPE]",
    "number": 12341234
  },
  "address": {
    "zip_code": "[FAKER][ADDRESS][ZIP_CODE]",
    "street_name": "[FAKER][ADDRESS][STREET_NAME]",
    "street_number": 2
  },
  "description": "Información del cliente",
  "date_created": "2021-05-25T15:36:23.541Z",
  "metadata": {},
  "cards": [
    {}
  ],
  "addresses": [
    {}
  ]
}
```

> Ejemplo de respuesta sin incluir el parámetro `customers_id`:

Reviewer: ```suggestion
Example of response without including the `customers_id` parameter:
```

```json
{
  "message": "missing customer id"
}
```
> NOTE
>

> > Nota

Reviewer: ```suggestion
> Note
```

>

> > Si no tienes el `customer_id`, [consulta la API de Clientes](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/reference/customers/_customers_search/get) y genera una solicitud `HTTP GET` utilizando el parámetro `email` para obtenerlo.

Reviewer: ```suggestion
> If you don't have the `customer_id`, [check the Customers API](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/en/reference/customers/_customers_search/get) and generate an` HTTP GET` request using the `email` parameter to get it.
```
